tools/autograd/load_derivatives.py

```diff
@@ -1,115 +1,115 @@
 # Parses derivatives.yaml into autograd functions
 #
-# Each autograd function is represented by dictionary containing a list of
-# derivatives (also a dictionary). See `create_autograd_function` and
-# `create_derivative` for the keys.
-from collections import defaultdict
-import copy
+# Each autograd function is represented by `DifferentiabilityInfo` containing
+# a list of `Derivative`. See `tools.codegen.api.autograd` for the data models.
+from collections import defaultdict, Counter
 import re
+from typing import Sequence, Any, Tuple, List, Set, Dict, Match, Optional
 import yaml
-from .utils import YamlLoader
-from .utils import IDENT_REGEX, split_name_params
 
-
-def load_derivatives(path, declarations):
-    with open(path, 'r') as f:
-        definitions = yaml.load(f, Loader=YamlLoader)
-
-    declarations_by_signature = defaultdict(list)
-    declarations_by_schema = dict()
-    for declaration in declarations:
-        declarations_by_signature[get_signature(declaration)].append(declaration)
-        if declaration['schema_string']:
-            assert declaration['schema_string'] not in declarations_by_schema
-            declarations_by_schema[declaration['schema_string']] = declaration
-
-    differentiability_infos = [
-        process_definition(defn, declarations_by_signature, declarations_by_schema)
+from tools.codegen.api.autograd import *
+from tools.codegen.api.types import *
+import tools.codegen.api.cpp as cpp
+from tools.codegen.gen import parse_native_yaml, with_native_function
+from tools.codegen.model import *
+from tools.codegen.utils import *
+
+try:
+    # use faster C loader if available
+    from yaml import CLoader as Loader
+except ImportError:
+    from yaml import Loader  # type: ignore
+
+def load_derivatives(derivatives_yaml_path: str, native_yaml_path: str) -> Sequence[DifferentiabilityInfo]:
+    with open(derivatives_yaml_path, 'r') as f:
+        definitions = yaml.load(f, Loader=Loader)
+
+    functions = parse_native_yaml(native_yaml_path)
+
+    # What's the difference between function schema v.s. signature?
+    # function schema is the complete declaration including mutability annotation / default value and etc.
+    # signature is the canonical schema for a group of functions (in/out/functional variants)
```

Comment:
What does the "in" variant means here?

Reply:
Changed it to 'in-place' - referencing the terms used in model.py